### PR TITLE
run from arbitrary directories

### PR DESCRIPTION
**Pull request by boriskaus**
This PR allows running the examples from arbitrary directories. This is handy as I now registered the package (should be merged in a few days) after which users can install it and run it locally with:
```Julia
julia>using InteractiveGeodynamics
julia> RisingSphere()
julia> Convection()
julia> RayleighTaylor()
```

To be added:
- [ ] change axes of the plot

### CODE REVIEW

File: RayleighBenardConvection/dash_functions_convection.jl [hunk 1]
```diff
@@ -86,7 +128,29 @@ function create_main_figure(
...
 end
+=#
 
+"""
+Creates a setup with noisy temperature and one phase
+"""
+function CreateSetup(ParamFile, ΔT=1000, ampl_noise=100; args)
+    Grid = ReadLaMEM_InputFile(ParamFile, args=args)
+    Phases = zeros(Int64, size(Grid.X))
+    Temp = ones(Float64, size(Grid.X)) * ΔT / 2
```
Comment (albert-de-montserrat) [inline RayleighBenardConvection/dash_functions_convection.jl]: These two lines could simply be 
```julia
Temp = [ΔT / 2 + rand()*ampl_noise for _ in axes(Grid.X,1), _ in axes(Grid.X,2)]
```

File: RayleighTaylorInstability/dash_functions_RTI.jl [hunk 2]
```diff
@@ -1,5 +1,90 @@
...
+    Phases      =   zeros(Int64, size(Grid.X));      
+    Temp        =   zeros(Float64,size(Grid.X));  
+    
+
+    if layered_overburden
+        H_layer = 0.25;
+        for z_low = minimum(Grid.Z):2*H_layer:maximum(Grid.Z)
+            # print(z_low)
+            # z_low = -z_low
+            iz = findall( (Grid.Z[1,1,:] .> z_low) .&  (Grid.Z[1,1,:] .<= (z_low+H_layer) ))
```
Comment (albert-de-montserrat) [inline RayleighTaylorInstability/dash_functions_RTI.jl]: there's no need to use `findall`, you can simply remove it and then `iz` will be a bitmask

File: RayleighTaylorInstability/dash_functions_RTI.jl [hunk 3]
```diff
@@ -1,5 +1,90 @@
...
+        H_layer = 0.25;
+        for z_low = minimum(Grid.Z):2*H_layer:maximum(Grid.Z)
+            # print(z_low)
+            # z_low = -z_low
+            iz = findall( (Grid.Z[1,1,:] .> z_low) .&  (Grid.Z[1,1,:] .<= (z_low+H_layer) )) 
+            Phases[:,:,iz] .= 1;
+        end 
+    end
+   
+    z_int       =   ones(Grid.nump_x)*Hi + rand(Grid.nump_x)*ampl_noise
```
Comment (albert-de-montserrat) [inline RayleighTaylorInstability/dash_functions_RTI.jl]: `z_int = [Hi + rand()*ampl_noise for _ in 1:Grid.nump_x]`

File: RayleighTaylorInstability/dash_functions_RTI.jl [hunk 4]
```diff
@@ -1,5 +1,90 @@
...
+    # print(z_int)
+    # z_int       =   -z_int
+    for ix=1:Grid.nump_x, iy=1:Grid.nump_y
+        iz = findall(Grid.Z[ix,iy,:] .< z_int[ix] )
```
Comment (albert-de-montserrat) [inline RayleighTaylorInstability/dash_functions_RTI.jl]: Can remove `findall`

File: RisingSphere/dash_functions_RisingSphere.jl [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    ])
+end
+
+"""
+Creates a setup with noisy temperature and one phase
+"""
+function CreateSetup(ParamFile, ΔT=1000, ampl_noise=100; args)
+    Grid = ReadLaMEM_InputFile(ParamFile, args=args)
+    Phases = zeros(Int64, size(Grid.X))
+    Temp = ones(Float64, size(Grid.X)) * ΔT / 2
```
Comment (albert-de-montserrat) [inline RisingSphere/dash_functions_RisingSphere.jl]: Can use a list comprehension as in the other model

File: src/dash_tools.jl [hunk 6]
```diff
@@ -378,9 +404,26 @@ function make_time_card()
 end
 
 """
-Retunrs a row containing a label, a tooltip and a filling box.
+Returns a row containing a label, a tooltip and a filling box.
 """
-function make_accordion_item(label::String="param", idx::String="id", msg::String="Message", value::Float64=1.0, min::Float64=1.0e-10, max::Float64=10_000.0)
+function make_accordion_item(label::String="param", idx::String="id", msg::String="Message", value::_T=1*one(_T), min=nothing, max=nothing) where _T <: Number
+    
+    if isa(_T, Float64)
```
Comment (albert-de-montserrat) [inline src/dash_tools.jl]: You could refactor this branching as something like 
```julia
 @inline _check_min(::Float64, ::Nothing) = 1e-10
 @inline _check_min(::Int64, ::nothing) = 2
 @inline _check_min(::T, x) where T = x
 @inline _check_max(::T, ::Nothing) where T = T(10_000)
 @inline _check_max(::T, x) where T = x

min = _check_min(_T, min)
max = _check_min(_T, max)
```

and get it all inlined

File: src/dash_tools.jl [hunk 7]
```diff
@@ -378,9 +404,26 @@ function make_time_card()
 end
 
 """
-Retunrs a row containing a label, a tooltip and a filling box.
+Returns a row containing a label, a tooltip and a filling box.
 """
-function make_accordion_item(label::String="param", idx::String="id", msg::String="Message", value::Float64=1.0, min::Float64=1.0e-10, max::Float64=10_000.0)
+function make_accordion_item(label::String="param", idx::String="id", msg::String="Message", value::_T=1*one(_T), min=nothing, max=nothing) where _T <: Number
```
Comment (albert-de-montserrat) [inline src/dash_tools.jl]: Perhaps is better using different names for `min` and `max`, they are both julia functions. Maybe `low` and `high`?

**Comment by albert-de-montserrat** [comment]
Just for the record, this branch freezes when calling `RisingSphere()` in Windows and WSL with 1.9.0. Works fine on mac with 1.9.3.

**Comment by albert-de-montserrat** [comment]
Perhaps the GUI models folders could be moved into `src/` since they are part of the main module.

**Comment by albert-de-montserrat** [comment]
Not sure if [this](https://github.com/JuliaGeodynamics/InteractiveGeodynamics.jl/blob/ed0af4c307fd3d964b06d29d8d964f1228f0c04e/src/utils.jl#L39) function is used often or not, but it is  type unstable because `values` is changing type (the size is a type in this case) inside a loop. Since `tb` is a `NTuple`, the compiler does actually have all the information needed, just need to be rewritten as e.g.

```julia
function get_values_textboxes(tb::NTuple{N, T}) where {N,T}
    values = ntuple(Val(N)) do i 
        Base.@_inline_meta
        textb = tb[i]
        parse(Float64,textb.displayed_string[])
    end
    return values
end
```

**Comment by boriskaus** [comment]
thanks for your comments @albert-de-montserrat - I have implemented them all & deleted obsolete files where appropriate

**Pull request merged**